monitoring/host.go

```diff
@@ -29,58 +29,93 @@ import (
 	"github.com/dustin/go-humanize"
 )
 
-const (
-	hostCheckerID = "cpu-ram"
-)
+// NewHostChecker returns a new checker to validate host environment
+func NewHostChecker(config HostConfig) health.Checker {
+	return &hostChecker{
+		HostConfig: config,
+		getMemory:  realGetMemory,
+		getCPU:     realGetCPU,
+	}
+}
 
-// HostChecker validates CPU and RAM requirements
-type HostChecker struct {
-	// MinCPU is minimum amount of logical CPUs
+// HostConfig describes host environment requirements
+type HostConfig struct {
+	// MinCPU specifies the minimum amount of logical CPUs to expect
 	MinCPU int
-	// MinRamBytes is minimum amount of RAM
+	// MinRAMBytes specifies the minimum amount of RAM to expect
 	MinRAMBytes uint64
 }
 
+// hostChecker validates CPU and RAM requirements
+type hostChecker struct {
+	HostConfig
+	getMemory memoryGetterFunc
+	getCPU    cpuGetterFunc
+}
+
 // Name returns checker id
-func (c *HostChecker) Name() string {
+// Implements health.Checker
+func (c *hostChecker) Name() string {
 	return hostCheckerID
 }
 
-// Check runs check
-func (c *HostChecker) Check(ctx context.Context, reporter health.Reporter) {
-	failed := false
+// Check validates that the host has enough RAM/CPUs.
+// Implements health.Checker
+func (c *hostChecker) Check(ctx context.Context, reporter health.Reporter) {
+	err := c.check(ctx, reporter)
+	if err != nil && !trace.IsNotFound(err) {
+		reporter.Add(NewProbeFromErr(c.Name(), "failed to validate host environment", err))
+		return
+	}
+	if reporter.NumProbes() != 0 {
+		return
+	}
+	reporter.Add(NewSuccessProbe(c.Name()))
+}
 
-	var mem sigar.Mem
-	err := mem.Get()
+func (c *hostChecker) check(ctx context.Context, reporter health.Reporter) error {
+	mem, err := c.getMemory()
 	if err != nil {
-		reporter.Add(NewProbeFromErr(hostCheckerID, "querying memory info", trace.Wrap(err)))
+		return trace.Wrap(err, "failed to query memory info")
 	}
 
 	if mem.Total < c.MinRAMBytes {
-		failed = true
 		reporter.Add(&pb.Probe{
-			Checker: hostCheckerID,
-			Detail: fmt.Sprintf("need minimum %s RAM, have %s",
+			Checker: c.Name(),
+			Detail: fmt.Sprintf("at least %s of RAM required, only %s are available",
 				humanize.Bytes(c.MinRAMBytes), humanize.Bytes(mem.Total)),
 			Status: pb.Probe_Failed,
 		})
 	}
 
-	if runtime.NumCPU() < c.MinCPU {
-		failed = true
+	if c.getCPU() < c.MinCPU {
 		reporter.Add(&pb.Probe{
-			Checker: hostCheckerID,
-			Detail:  fmt.Sprintf("need minimum %d CPU, have %d", c.MinCPU, runtime.NumCPU()),
-			Status:  pb.Probe_Failed,
+			Checker: c.Name(),
+			Detail: fmt.Sprintf("at least %d CPUs required, only %d are available",
```

Comment:
If this is a 1-core machine, the message will be grammatically incorrect, "only 1 are available". I would rephrase to avoid is/are, e.g. "have only X".